fuse_constraints/test/test_variable_constraints.cpp

```diff
@@ -101,6 +101,23 @@ TEST(VariableConstraints, GetConstraints)
   EXPECT_EQ(expected1, actual1);
   EXPECT_EQ(expected2, actual2);
   EXPECT_EQ(expected3, actual3);
+
+  auto actual0_iter = actual0.begin();
+  actual0_iter = vars.getConstraints(0u, actual0_iter);
+
+  auto actual1_iter = actual1.begin();
+  actual1_iter = vars.getConstraints(1u, actual1_iter);
+
+  auto actual2_iter = actual2.begin();
+  actual2_iter = vars.getConstraints(2u, actual2_iter);
+
+  auto actual3_iter = actual3.begin();
+  actual3_iter = vars.getConstraints(3u, actual3_iter);
+
+  EXPECT_EQ(expected0.size(), std::distance(actual0.begin(), actual0_iter));
```

Comment:
Trying with `static_cast<std::ptrdiff_t>`, which I believe is the return type of `std::distance`.

I refrained from using `decltype` because the expression would be large, since `std::distance` is templated and I think the return type depends on the template argument.